metricbeat/helper/server/http/http.go

```diff
@@ -19,40 +19,24 @@ package http
 
 import (
 	"context"
-	"io/ioutil"
 	"net"
 	"net/http"
 	"strconv"
 
-	"github.com/elastic/beats/v7/libbeat/common"
 	"github.com/elastic/beats/v7/libbeat/common/transport/tlscommon"
 	"github.com/elastic/beats/v7/libbeat/logp"
 	"github.com/elastic/beats/v7/metricbeat/helper/server"
 	"github.com/elastic/beats/v7/metricbeat/mb"
 )
 
 type HttpServer struct {
-	server     *http.Server
-	ctx        context.Context
-	stop       context.CancelFunc
-	done       chan struct{}
-	eventQueue chan server.Event
+	server *http.Server
+	ctx    context.Context
+	stop   context.CancelFunc
+	done   chan struct{}
 }
 
-type HttpEvent struct {
-	event common.MapStr
-	meta  server.Meta
-}
-
-func (h *HttpEvent) GetEvent() common.MapStr {
-	return h.event
-}
-
-func (h *HttpEvent) GetMeta() server.Meta {
-	return h.meta
-}
-
-func NewHttpServer(mb mb.BaseMetricSet) (server.Server, error) {
+func NewHttpServer(mb mb.BaseMetricSet, handlerFunc func(writer http.ResponseWriter, req *http.Request)) (server.Server, error) {
```

Comment:
Use `http.HandlerFunc` already here?
```suggestion
func NewHttpServer(mb mb.BaseMetricSet, handlerFunc http.HandlerFunc) (server.Server, error) {
```